redhat-appstudio/service-provider-integration-operator
# feat: enable the quay and github service providers configuration as o…

Signed-off-by: Pavol Baran <pbaran@redhat.com>

### What does this PR do?
Allows users to create Quay or GitHub spiaccesstokenbinding without having Quay or GitHub service providers configured in oauth configuration secret. In this way, the binding is linked to an spiaccesstoken that is compatible only with manual token upload (and no OAuth flow).

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes https://issues.redhat.com/browse/SVPI-155

### How to test this PR?

1. Deploy operator with image: quay.io/pbaran/spio:SVPI-155
2. Delete Quay and GitHub service providers from oauth configuration secret (restart operator if needed to reload config)
3. Create binding `kubectl apply -f samples/spiaccesstokenbinding-2.yaml` and check that it is linked to spiaccesstoken with service-provider-type: Quay
4. Create binding `kubectl apply -f samples/spiaccesstokenbinding-3.yaml` and check that it is linked to spiaccesstoken with service-provider-type: GitHub
